msal/tests/Test.MSAL.NET.Integration/UsernamePasswordIntegrationTests.cs

```diff
@@ -0,0 +1,139 @@
+using Test.MSAL.NET.Integration.Infrastructure;
+
+namespace Test.MSAL.NET.Integration
+{
+    [TestClass]
+    public class UsernamePasswordIntegrationTests
+    {
+        public const string ClientId = "0615b6ca-88d4-4884-8729-b178178f7c27";
+        public const string Authority = "https://login.microsoftonline.com/organizations/";
+        public string[] Scopes = { "user.read" };
+        AuthHelper authHelper = new AuthHelper();
+
+        [TestMethod]
+        [TestCategory("UsernamePasswordIntegrationTests")]
+        public async Task AcquireTokenWithManagedUsernamePasswordAsync()
+        {
+            //Get User from Lab
+            var user = authHelper.GetUser(
+                new UserQueryParameters
+                {
+                    IsMamUser = false,
+                    IsMfaUser = false,
+                    IsFederatedUser = false
+                });
+
+            SecureString securePassword = new NetworkCredential("", ((LabUser)user).GetPassword()).SecurePassword;
+
+            PublicClientApplication msalPublicClient = new PublicClientApplication(ClientId, Authority);
+            
+            AuthenticationResult authResult = await msalPublicClient.AcquireTokenByUsernamePasswordAsync(Scopes, user.Upn, securePassword);
+            Assert.IsNotNull(authResult);
+            Assert.IsNotNull(authResult.AccessToken);
+            Assert.AreEqual(user.Upn, authResult.Account.Username);
+        }
+
+        [TestMethod]
+        [TestCategory("UsernamePasswordIntegrationTests")]
+        public async Task AcquireTokenWithFederatedUsernamePasswordAsync()
+        {
+            //Get User from Lab
+            var user = authHelper.GetUser(
+                new UserQueryParameters
+                {
+                    IsMamUser = false,
+                    IsMfaUser = false,
+                    IsFederatedUser = true
+                });
+
+            SecureString securePassword = new NetworkCredential("", ((LabUser)user).GetPassword()).SecurePassword;
+
+            PublicClientApplication msalPublicClient = new PublicClientApplication(ClientId, Authority);
+            AuthenticationResult authResult = await msalPublicClient.AcquireTokenByUsernamePasswordAsync(Scopes, user.Upn, securePassword);
+            Assert.IsNotNull(authResult);
+            Assert.IsNotNull(authResult.AccessToken);
+            Assert.AreEqual(user.Upn, authResult.Account.Username);
+        }
+
+        [TestMethod]
+        [TestCategory("UsernamePasswordIntegrationTests")]
+        public void AcquireTokenWithManagedUsernameIncorrectPassword()
+        {
+            //Get User from Lab
+            var user = authHelper.GetUser(
+                new UserQueryParameters
+                {
+                    IsMamUser = false,
+                    IsMfaUser = false,
+                    IsFederatedUser = false
+                });
+
+            SecureString securePassword = new SecureString();
```

Comment:
rename variable to incorrectSecurePassword, then you can remove the comment you added below